cmake/TargetLinkLibrariesSystem.cmake

```diff
@@ -5,18 +5,7 @@ function(target_link_libraries_system target)
   set(libs ${ARGN})
   foreach(lib ${libs})
     get_target_property(lib_include_dirs ${lib} INTERFACE_INCLUDE_DIRECTORIES)
-    target_include_directories(${target} SYSTEM PRIVATE ${lib_include_dirs})
-    target_link_libraries(${target} ${lib})
+    target_include_directories(${target} SYSTEM PUBLIC ${lib_include_dirs})
```

Comment:
Do you know what the effects of public include dirs are? Not getting what you changed here.

Reply:
PUBLIC makes it visible to whoever includes `target`. As such, whoever includes `hyrise_impl` automatically gets the json library.